src/lib/notifier.c

```diff
@@ -109,6 +109,9 @@ void notifier_unregister(void *receiver, void *caller, enum notify_id type)
 		    (!caller || handle->caller == caller)) {
 			if (!--handle->num_registrations) {
 				list_item_del(&handle->list);
+				#ifdef CONFIG_PROBE
+					list_item_del(&notify->list[NOTIFIER_ID_BUFFER_PRODUCE]);
+				#endif
```

Comment:
Presumably this would be a bug fix. Are calls to
```
	notifier_unregister(_probe, dev->cb, NOTIFIER_ID_BUFFER_PRODUCE);
```
in probe.c not enough?

If this is indeed needed, then please move `#ifdef` / `#endif` to position 0, and reduce indentation of `list_item_del()`